documentation/library_guide/dynamic_selection_api/backends.rst

```diff
@@ -0,0 +1,162 @@
+Backends
+########
+
+The dynamic selection API is an experimental feature in the |onedpl_long|  
+(|onedpl_short|) that selects an *execution resource* based on a chosen  
+*selection policy*. For the policies to work with different resource types.  
+the resource management and work submission mechanics are handled  
+separately by resource-specific *backends*.
+
+A backend is responsible for:  
+
+1. **Resource Management** - storing and providing access to a set of resources  
+2. **Work Submission** - executing user functions on selected resources  
+3. **Synchronization** - waiting for submitted work to complete  
+4. **Instrumentation** - optionally reporting execution information (timing, completion)  
+
+Backends are typically not directly visible to application developers - they work  
+behind the scenes to enable policies to function.
+
+SYCL Backend
+------------
+
+A sycl backend is provided to manage ``sycl::queue`` core resources, and ``sycl::queue``
+is the default resource if none is provided or deduce-able from resource arguments.
+The SYCL backend provides:
+
+- Default initialization from available SYCL devices
+- Event-based submission and waiting
+- Optional profiling and timing instrumentation
+
+When you construct a policy without specifying resources, it uses the SYCL
+backend's default initialization:
+
+.. code:: cpp
+
+  #include <oneapi/dpl/dynamic_selection>
+
+  namespace ex = oneapi::dpl::experimental;
+
+  // Uses SYCL backend with default device initialization
+  ex::round_robin_policy<> policy;
+
+  // Explicitly specifying SYCL queues
+  std::vector<sycl::queue> my_queues = { /* ... */ };
+  ex::round_robin_policy<sycl::queue> policy2{my_queues};
+
+The SYCL backend supports the reporting for :ref:`Execution Information <execution-information>`
+of ``task_submission``, ``task_completion``, and ``task_time``.
+
+Backend Architecture
+--------------------
+
+The backend system uses a layered design to support multiple resource types:
+
+**default_backend**: When you create a policy, ``default_backend`` automatically
+determines the core resource type by applying any
+:ref:`resource adapter <resource-adapters>` to your resource type. It then 
+delegates to the appropriate ``core_resource_backend`` specialization.
+
+**core_resource_backend**: Specialized backend implementations exist for specific core
+resource types (like ``sycl::queue``). A generic implementation is provided
+for core resources without an explicitly specialized implementation which provides
+a minimal amount of functionality:
+
+- Basic resource storage and retrieval
+- Simple work submission without instrumentation (No execution information reporting)
+
+To use a custom resource type with full instrumentation support, you must create a
+``core_resource_backend`` specialization.
+
+Lazy Reporting
+--------------
```

Comment:
I wonder if we should name it "on-demand reporting" or similarly - at least in the documentation (while the API naming can be discussed / addressed later).

Reply:
@vossmjp Do you have an opinion here?  This is not something new in the design, but just something we've documented.  I have no strong opinions here, but there are a few places we would have to change it if we choose to.